test/project-name-service.ts

```diff
@@ -0,0 +1,90 @@
+/// <reference path=".d.ts" />
+"use strict";
+
+import * as yok from "../lib/common/yok";
+import { ProjectNameService } from "../lib/services/project-name-service";
+import {assert} from "chai";
+import {ErrorsStub, LoggerStub} from "./stubs";
+
+let mockProjectNameValidator = {
+	validate: () => { return true; }
+};
+
+let dummyString: string = "dummyString";
+
+function createTestInjector(): IInjector {
+	let testInjector: IInjector;
+
+	testInjector = new yok.Yok();
+	testInjector.register("projectNameService", ProjectNameService);
+	testInjector.register("projectNameValidator", mockProjectNameValidator);
+	testInjector.register("errors", ErrorsStub);
+	testInjector.register("logger", LoggerStub);
+	testInjector.register("prompter", {
+		confirm: (message: string): IFuture<boolean> => {
+			return (() => {
+				return true;
+			}).future<boolean>()();
+		},
+		getString: (message: string): IFuture<string> => {
+			return (() => {
+				return dummyString;
+			}).future<string>()();
+		}
+	});
+
+	return testInjector;
+}
+
+describe("Project Name Service Tests", () => {
+	let testInjector: IInjector;
+	let projectNameService: IProjectNameService;
+	let validProjectName = "valid";
+	let invalidProjectName = "1invalid";
+
+	before(() => {
```

Comment:
before is called before the first test only. We prefer using `beforeEach`, so each test will be on clean setup